src/BDCS/API/Recipes.hs

```diff
@@ -579,6 +584,188 @@ printOId oid =
     Git.oIdToString oid >>= print
 
 
+-- | Type of Diff Entry
+--
+-- Used by RecipeDiffEntry's old and new fields
+data RecipeDiffType =
+    Name {rdtName :: String}
+  | Description {rdtDescription :: String}
+  | Version {rdtVersion :: Maybe String}
+  | Module {rdtModule :: RecipeModule}
+  | Package {rdtPackage :: RecipeModule}
+  | None
+  deriving (Eq, Show)
+
```

Comment:
You don't have to create data types that are records.  If you think these are likely to only ever be one element each, you can make them like so:
```
data RecipeDiffType = 
    Name String
  | Description String
...
```